src/pks/flow/constitutive_relations/sources/field_mass_source_evaluator.hh

```diff
@@ -0,0 +1,66 @@
+/*
+  Copyright 2010-202x held jointly by participating institutions.
+  ATS is released under the three-clause BSD License.
+  The terms of use and "as is" disclaimer for this license are
+  provided in the top-level COPYRIGHT file.
+
+  Authors: Phong V.V. Le
+*/
+
+//! Evaluates tile source mass from tile discharge
+/*!
+
+Mass sources into stream/river due to tile discharge
+
+.. _source_mass-tiles-spec:
+.. admonition:: source_mass-tiles-spec
+*/
```

Comment:
Needs some documentation, and the name should match the file, eg. `qc-relation-evaluator-spec` or similar.  At least the `"function`" list needs to be added, and KEYS, so something like:

```
.. _qc-relation-evaluator-spec:
.. admonition:: qc-relation-evaluator-spec

   `"function`" ``[function-spec]`` Function describing the relationship between discharge/water mass and concentration of the incoming water.

   KEYS:
      - `"cell volume`" **DOMAIN-cell_volume**
      - `"molar density liquid`" **DOMAIN-molar_density_liquid**
      - `"field mass source`" **DOMAIN-field_mass_source**
```